modules/reindex/src/main/java/org/elasticsearch/reindex/AbstractAsyncBulkByScrollAction.java

```diff
@@ -256,6 +256,8 @@ protected BulkRequest buildBulk(Iterable<? extends ScrollableHitSource.Hit> docs
     }
 
     protected ScrollableHitSource buildScrollableResultSource(BackoffPolicy backoffPolicy) {
+        // Do not open scroll if maxDocs <= scroll size
+        mainRequest.disableScrollIfUnnecessary();
```

Comment:
May be this could actually be inlined here to avoid introducing additional methods into request

Reply:
Just realized this is not working for `org.elasticsearch.reindex.Reindexer.AsyncIndexBySearchAction#buildScrollableResultSource` as this constructs `RemoteScrollableHitSource` with `mainRequest.getSearchRequest()` without calling super. I will probably move this logic to constructor before `buildScrollableResultSource` is called